### DISCUSSION TITLE
"-9997 Invalid sample rate" seems like an impossible error on pulseaudio unless Audacity did something rather wrong

### DISCUSSION THREAD

**Discussion by ell1e** [audacity/audacity#4695] [Q&A]
### Bug description

"-9997 Invalid sample rate" seems like an impossible error on pulseaudio unless Audacity did something rather wrong. This is what I'm getting on my ARM64 device with pulseadio, and since pulseaudio can resample I assume Audacity did something very wrong  opening the audio device (like picking some absurd sample rate not even puleaudio can resample to).

### Steps to reproduce

1. Open Audacity with new project
2. Click Transport > Recording > Record in the menu
3. Error message about invalid sample rate appears

### Expected behavior

Recording works or I get an error that doesn't seem impossible to be the actual problem

### Actual behavior

An error message shows up that doesn't really seem plausible if Audacity had done something reasonable

### Audacity Version

other

### Operating system

Linux

### Additional context

I failed to find out the exact version:

```
$ flatpak run org.audacityteam.Audacity --version

(process:2): Gdk-CRITICAL **: 02:59:32.084: gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:2): Gdk-CRITICAL **: 02:59:32.084: gdk_window_get_display: assertion 'GDK_IS_WINDOW (window)' failed

(process:2): Gdk-CRITICAL **: 02:59:32.085: gdk_cursor_new_from_pixbuf: assertion 'GDK_IS_DISPLAY (display)' failed
```
(On this device I can't reach the about menu entry easily to check there.)

**Comment by LWinterberg** [comment]
The error codes aren't always super accurate to what's going on on a user level; this error for example also can occur if overdub is turned on and the device can't handle that, and if you're asking to record a number of channels that it doesn't support
